flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/snapshot/ForStNativeFullSnapshotStrategy.java

```diff
@@ -152,12 +153,14 @@ public ForStNativeSnapshotResources syncPrepareResources(long checkpointId) thro
         final PreviousSnapshot previousSnapshot =
                 snapshotMetaData(checkpointId, stateMetaInfoSnapshots);
 
-        // Disable file deletion for file transformation. ForSt will decide whether to allow file
-        // deletion based on the number of calls to disableFileDeletions() and
-        // enableFileDeletions(), so disableFileDeletions() should be call only once.
-        db.disableFileDeletions();
+        try (ResourceGuard.Lease ignoredLease = resourceGuard.acquireResource()) {
```

Comment:
I'd suggest we acquire lease here, and release this one after the snapshot finished (below `enableFileDeletions `).

Reply:
👍 I changed it to only acquire once.